Sources/GRPC/GRPCServerPipelineConfigurator.swift

```diff
@@ -103,18 +103,29 @@ final class GRPCServerPipelineConfigurator: ChannelInboundHandler, RemovableChan
       channel: channel,
       targetWindowSize: self.configuration.httpTargetWindowSize
     ) { stream in
-      // TODO: use sync options when NIO HTTP/2 support for them is released
-      // https://github.com/apple/swift-nio-http2/pull/283
-      stream.getOption(HTTP2StreamChannelOptions.streamID).map { streamID -> Logger in
-        logger[metadataKey: MetadataKey.h2StreamID] = "\(streamID)"
-        return logger
-      }.recover { _ in
-        logger[metadataKey: MetadataKey.h2StreamID] = "<unknown>"
-        return logger
-      }.flatMap { logger in
+      let streamID: Int?
+
+      // Sync options were added to the HTTP/2 stream channel in 1.17.0 (we require at least this)
+      // so this shouldn't be `nil`, but not a problem if it is.
+      if let syncOptions = stream.syncOptions {
+        do {
+          streamID = Int(try syncOptions.getOption(HTTP2StreamChannelOptions.streamID))
```

Comment:
Nit: As you don't care about the error you could condense the code by using `try?` with a `flatmap`.

Reply:
Good idea!